Marlin/Marlin_main.cpp

```diff
@@ -5922,18 +5929,17 @@ inline void gcode_M17() {
   }
 
   static void ensure_safe_temperature() {
-    bool did_show = false;
     wait_for_heatup = true;
     while (wait_for_heatup) {
       idle();
+      if (!wait_for_heatup) break;
```

Comment:
This can never happen. The `while` statement will already have caught this condition.